### ISSUE TITLE
[resilience4j-spring-boot2] Set register-health-indicator property to true by default

### ISSUE THREAD

**Issue by JoseLora** [resilience4j/resilience4j#1554]
Resilience4j version: 1.7.0

Java version: openjdk-11.0.12-1

Hello.

We'd like to set `register-health-indicator` property to true by default for every circuit-breaker defined in yaml, but to do that, we need to define a baseConfig and made all circuit-breakers inherit from that config using `baseConfig` property.

Would it be possible to define some kind of special `defaultConfig` that is always inherited by all circuit-breakers without needing to set  `baseConfig` property for each?

We develop a internal company framework on top of spring-boot and sometimes it's very useful for us to provide some default config values for all our users (internal application developers).

If this is not possible... would you please make this method protected so we could easily extend it and modify its behaviour?

https://github.com/resilience4j/resilience4j/blob/7ce1227f9d120fd5175ca9abf9927a26d9fbabfc/resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/health/CircuitBreakersHealthIndicator.java#L100-L104

Thanks!

**Comment by RobWin** [comment]
Hi,

do other circuitbreaker instances use the default config as-is or do you plan to override the default config?

**Comment by JoseLora** [comment]
Hi,

At the moment, we can define configs and inherit from them using `baseConfig` property.

```
resilience4j.circuitbreaker:
    configs:
        default:
            slidingWindowSize: 100
            permittedNumberOfCallsInHalfOpenState: 10
            waitDurationInOpenState: 10000
            failureRateThreshold: 60
            eventConsumerBufferSize: 10
            registerHealthIndicator: true
        someShared:
            slidingWindowSize: 50
            permittedNumberOfCallsInHalfOpenState: 10
    instances:
        backendA:
            baseConfig: default
            waitDurationInOpenState: 5000
        backendB:
            baseConfig: someShared
```

The idea could be to define some special config that will be inherited by default without the need to specify `baseConfig` for each instance.
```
resilience4j.circuitbreaker:
    configs:
        SPECIAL_DEFAULT_CONFIG_INHERITED_BY_DEFAULT:
            registerHealthIndicator: true
    instances:
        backendA:
            waitDurationInOpenState: 5000
        backendB:
            waitDurationInOpenState: 7000
```
In example above, backendA and backendB will have registerHealthIndicator=true automatically.

But, we could also override that default config:
```
resilience4j.circuitbreaker:
    configs:
        SPECIAL_DEFAULT_CONFIG_INHERITED_BY_DEFAULT:
            registerHealthIndicator: true
    instances:
        backendA:
            waitDurationInOpenState: 5000
            registerHealthIndicator: false
        backendB:
            waitDurationInOpenState: 7000
```
In this case, backendA would not register health indicator.
 
Basically, we develop an internal company framework and we'd like to change the default value for some properties, in our case, register-health-indicator=true by default.

If we can't do this by yaml, we are trying to extend and override default `CircuitBreakersHealthIndicator` but all internal methods are private. It would be great to set at least `isRegisterHealthIndicator` as protected so we could easily extend `CircuitBreakersHealthIndicator` and only override that method.

Maybe you know a better way to achieve this. Thanks in advance!
